system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -43,8 +42,60 @@ ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSet
     PeriodicMetricSink(name, "elastic", pSettingsTree)
 {
     ignoreZeroMetrics = pSettingsTree->getPropBool("@ignoreZeroMetrics", true);
-    pSettingsTree->getProp("@elasticHost", elasticHost);
-    pSettingsTree->getProp("@indexName", indexName);
+
+    StringBuffer hostName;
+    StringBuffer hostProtocol;
+    StringBuffer hostPort;
+
+    Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
+    if (pHostConfigTree)
+    {
+        pHostConfigTree->getProp("@name", hostName);
```

Comment:
minor future point:  If the tree is guaranteed to stay active then more common is to use queryProp() and avoid cloning the strings.  In this case the code is only executed once, so it doesn't matter.

```
const char * protocol = pHostConfigTree->queryProp("@protocol");
if (!protocol)
   protocol = "https";
   
```